pkg/hydrate/for_each_cluster.go

```diff
@@ -46,23 +61,21 @@ const (
 // - err, the MultiError which Parser.Parse returned, if there was one.
 //
 // Per standard ForEach conventions, ForEachCluster has no return value.
-func ForEachCluster(parser filesystem.ConfigParser, options validate.Options,
-	sourceFormat filesystem.SourceFormat, filePaths reader.FilePaths,
-	f func(clusterName string, fileObjects []ast.FileObject, err status.MultiError)) {
-	clusterRegistry, errs := parser.ReadClusterRegistryResources(filePaths, sourceFormat)
+func ForEachCluster(parseOpts ParseOptions, validateOpts validate.Options, f ClusterFilterFunc) {
+	clusterRegistry, errs := parseOpts.Parser.ReadClusterRegistryResources(parseOpts.FilePaths, parseOpts.SourceFormat)
 	clustersObjects, err2 := selectors.FilterClusters(clusterRegistry)
-	clusterNames, err3 := parser.ReadClusterNamesFromSelector(filePaths)
+	clusterNames, err3 := parseOpts.Parser.ReadClusterNamesFromSelector(parseOpts.FilePaths)
 	errs = status.Append(errs, err2, err3)
```

Comment:
The different error variables seem to just add confusion, at least for me. Can this be refactored to just use a single `err` variable and call `status.Append(errs, err)` after each assignment?

Reply:
done